Need to update `cryptography` module version for running on Apple Silicon M1 architecture.

**Environment**
- How did you install GRR? : **git clone from source**
- What GRR version are you running?: **3.4.5.1**
- What operating system does the GRR server run on? : macOS Monterey (12.3.1) - Apple Silicon M1 

**Describe the issue**

Hello, everyone in the community. 🙂
The issue is derived from the [dftimewolf](https://github.com/log2timeline/dftimewolf) project associated with the GRR project.
To enable and debug dftimewolf, you must install `grr_api_client` for dependencies.

Referring to the code below, `grr_api_client` is forcing the version of cryptography you use (2.9.2)
(The comments in the code also state TODO that you need to upgrade.)

https://github.com/google/grr/blob/7c9ac0c9cc793cdd696a612ccad938ccf57baa02/api_client/python/setup.py#L69

However, in an Apple Silicon M1 environment, that version is experiencing errors, and this requires that the `cryptography` version be kept at least `3.3.1` or higher.

Since the `grr_api_client` forced the cryptographic version, it is difficult to resolve the error in "dftimewolf".
If possible, we need to upgrade the version or allow the version above it.



**Error logs**
```shell
> pip3 install -r requirements.txt
```

```
               ^
      build/temp.macosx-10.14-arm64-cpython-38/_openssl.c:20782:10: note: did you mean 'ERR_GET_LIB'?
      /opt/homebrew/opt/openssl@3/include/openssl/err.h:241:36: note: 'ERR_GET_LIB' declared here
      static ossl_unused ossl_inline int ERR_GET_LIB(unsigned long errcode)
                                         ^
      build/temp.macosx-10.14-arm64-cpython-38/_openssl.c:20798:14: error: implicit declaration of function 'ERR_GET_FUNC' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        { result = ERR_GET_FUNC(x0); }
                   ^
      build/temp.macosx-10.14-arm64-cpython-38/_openssl.c:25342:10: error: implicit declaration of function 'FIPS_mode' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        return FIPS_mode();
               ^
      build/temp.macosx-10.14-arm64-cpython-38/_openssl.c:25353:14: error: implicit declaration of function 'FIPS_mode' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        { result = FIPS_mode(); }
                   ^
      build/temp.macosx-10.14-arm64-cpython-38/_openssl.c:25368:10: error: implicit declaration of function 'FIPS_mode_set' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        return FIPS_mode_set(x0);
               ^
      fatal error: too many errors emitted, stopping now [-ferror-limit=]
      4 warnings and 20 errors generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for cryptography
Failed to build cryptography
ERROR: Could not build wheels for cryptography, which is required to install pyproject.toml-based projects
```

**Additional context**
If there is any other reason or solution for this error, please feel free to leave a thread. I'll try to solve it.

--- a/api_client/python/setup.py
+++ b/api_client/python/setup.py
@@ -64,9 +64,7 @@ setup_args = dict(
     },
     install_requires=[
         "grr_response_proto==%s" % VERSION.get("Version", "packagedepends"),
-        # TODO: This should be 3.3.2, but AppVeyor has issues with
-        # that version.
-        "cryptography==2.9.2",
+        "cryptography==3.3.2",
         "requests==2.25.1",
         "Werkzeug>=0.16.0,<=1.0.1",
     ],

--- a/grr/core/setup.py
+++ b/grr/core/setup.py
@@ -133,9 +133,7 @@ setup_args = dict(
         "sdist": Sdist,
     },
     install_requires=[
-        # TODO: This should be 3.3.2, but AppVeyor has issues with
-        # that version.
-        "cryptography==2.9.2",
+        "cryptography==3.3.2",
         "distro==1.5.0",
         "fleetspeak==0.1.9",
         "grr-response-proto==%s" % VERSION.get("Version", "packagedepends"),

